strawberryfields/engine.py

```diff
@@ -127,870 +78,173 @@
-            raise ValueError(
-                'Identity transformation only takes one parameter.')
-
-    def __str__(self):
-        """Prints the RegRefTransform using Blackbird syntax."""
-        temp = [str(r) for r in self.regrefs]
-        rr = ', '.join(temp)
-
-        if len(temp) > 1:
-            rr = '[' + rr + ']'
-
-        if self.func is None:
-            return 'RR({})'.format(rr)
-
-        if self.func_str is None:
-            return 'RR({}, {})'.format(rr, self.func.__name__)
-
-        return 'RR({}, {})'.format(rr, self.func_str)
-
-    def __format__(self, format_spec):
-        return self.__str__()  # pragma: no cover
-
-    def __eq__(self, other):
-        "Not equal to anything."
-        return False
-
-    def evaluate(self):
-        """Evaluates the numeric value of the function if all the measurement values are available.
-
-        Returns:
-            Number: function value
-        """
-        temp = [r.val for r in self.regrefs]
-        if any(v is None for v in temp):
-            # NOTE: "if None in temp" causes an error if temp contains arrays,
-            # since it uses the == comparison in addition to "is"
-            raise CircuitError("Trying to use a nonexistent measurement result "
-                               "(e.g. before it can be measured).")
-        if self.func is None:
-            return temp[0]
-        return self.func(*temp)
+from .backends.base import (NotApplicableError, BaseBackend)
 
 
 class Engine:
-    r"""Quantum compiler engine.
-
-    Acts as a context manager (and the context itself) for quantum circuits.
-    The contexts may not be nested.
-
-    .. currentmodule:: strawberryfields.engine.Engine
-
-    The quantum circuit is inputted by using the :meth:`~strawberryfields.ops.Operation.__or__`
-    methods of the quantum operations, which call the :meth:`append` method of the Engine.
-    :meth:`append` checks that the register references are valid and then
-    adds a new :class:`.Command` instance to the Engine command queue.
+    r"""Quantum program executor engine.
 
-    :meth:`run` executes the command queue on the chosen backend, and makes
+    :meth:`run` executes :class:`.Program` instances on the chosen backend, and makes
     measurement results available via the :class:`.RegRef` instances.
 
-    The ``New`` and ``Del`` operations modify the quantum register itself by adding
-    and deleting subsystems. The Engine keeps track of these changes as
-    they enter the command queue in order to be able to report register
-    reference errors as soon as they happen.
-
-    The backend, however, only becomes aware of subsystem changes when
-    the circuit is run. See :meth:`reset_queue` and :meth:`reset`.
-
     Args:
-        num_subsystems (int): Number of subsystems in the quantum register.
+        backend (str): name of the backend
+        host    (str): URL of the remote backend host, or None if a local backend is used
```

Comment:
Since this existing `Engine` class will likely become `LocalEngine`, the `host` argument is not needed, right?

Reply:
Right.